server/src/controllers/contactController.ts

```diff
@@ -373,56 +342,61 @@ contactController.get('/orders/:id', async (req: Request, res: Response) => {
       return;
     }
 
-    const {orders, donations, ...remainderOfContact} = contact;
-
-    const flattenedOrders = contact.orders.map((order) => {
-      const orderItems = order.orderitems.map((item) => {
-        const singleTickets = item.singletickets.filter((ticket) => !ticket.ticketwasswapped);
-        const quantity = singleTickets.length;
-        const ticketInfo = singleTickets.map((ticket) => {
-          if (!ticket.eventtickets.length) return null;
-          return {
-            description: ticket.eventtickets[0].eventinstances.events.eventname,
-            eventdate: ticket.eventtickets[0].eventinstances.eventdate,
-            eventtime: ticket.eventtickets[0].eventinstances.eventtime,
-            eventname: ticket.eventtickets[0].eventinstances.events.eventname,
-            detail: ticket.eventtickets[0].eventinstances.detail,
-            seasonname: ticket.eventtickets[0].eventinstances.events.seasons?.name,
-            tickettype: ticket.eventtickets[0].ticketrestrictions?.tickettype.description,
-          };
-        }).filter((ticket) => ticket !== null);
-
-        if (!ticketInfo[0]) return null;
-        return {
-          price: item.price,
-          quantity: quantity,
-          description: ticketInfo[0].description,
-          eventdate: ticketInfo[0].eventdate,
-          eventtime: ticketInfo[0].eventtime,
-          eventname: ticketInfo[0].eventname,
-          seasonname: ticketInfo[0].seasonname,
-          tickettype: ticketInfo[0].tickettype,
-          detail: ticketInfo[0].detail,
-        };
-      }).filter((item) => item !== null);
-
-      return {
+    const {orders, ...remainderOfContact} = contact;
+    const formattedDonations: any[] = [];
+    const flattenedOrders : any[] = [];
+    contact.orders.forEach((order) => {
+      const orderItemsMap = new Map<string, any>();
+      const {
+        ordertotal,
+        refunded,
+      } = order
+          .order_ticketitems
+          .reduce<{ordertotal: number, refunded: boolean}>((acc, ticket) => {
+            if (!ticket.ticketitem) return acc;
+            const key = `${ticket.ticketitem.ticketrestriction.eventinstanceid_fk}T${ticket.ticketitem.ticketrestriction.tickettypeid_fk}`;
```

Comment:
You may also want to add the price to the key, there is a case where an admin may want to purchase two of the same ticket types for the same event, but set their price differently.

Reply:
Fixed